src/Domain/Domain.hpp

```diff
@@ -53,9 +53,14 @@ class Domain {
    * [domain creation tutorial](\ref tutorial_domain_creation).
    *
    */
-  explicit Domain(std::vector<std::unique_ptr<domain::CoordinateMapBase<
-                      Frame::Logical, Frame::Inertial, VolumeDim>>>
-                      maps) noexcept;
+  explicit Domain(
+      std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Logical, Frame::Inertial, VolumeDim>>>
+          maps,
+      std::vector<DirectionMap<
+          VolumeDim,
+          std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>
+          boundary_conditions = {}) noexcept;
```

Comment:
same question about the necessity of this default. Also it would probably be good to add a note on the "specify all b.c. or none at all" requirement to the docs for this class.